README.md

```diff
@@ -54,6 +63,403 @@ npm install @dojo/shim
+```
+
+##### Global Error Outlet
+
+Whenever a `MatchType.ERROR` occurs a global outlet is automatically added to the matched outlets called `errorOutlet`. This outlet can be used to render a widget for any unknown routes.
+
+```ts
+const ErrorOutlet = Outlet(ErrorWidget, 'errorOutlet');
+```
+
+#### Route Registration
+
+Routes are registered using `RouteConfig`, that defines a route's `path`, the associated `outlet` and nested child `RouteConfig`s. The full routes are recursively constructed from the nested route structure.
+
+Example routing configuration:
+
+```ts
+const config: RouteConfig[] = [
+	{
+		path: 'foo',
+		outlet: 'root',
+		children: [
+			{
+				path: 'bar',
+				outlet: 'bar'
+			},
+			{
+				path: 'baz',
+				children: [
+					{
+						path: 'qux'
+					}
+				]
+			}
+		]
+	}
+]
+```
+
+That would register the following routes and outlets:
+
+| Route        | Outlet |
+| ------------ | ------ |
+|`/foo`        | `root` |
+|`/foo/bar`    | `bar`  |
+|`/foo/baz`    | `baz`  |
+|`/foo/baz/qux`| `qux`  |
+
+**Note:** If an `outlet` is not explicitly specified the `path` will be used.
+
+To actually register the configuration with a Dojo 2 router, simply pass the config object in the `constructor` options
+
+```ts
+const router = new Router({ config });
+```
+
+For routes that have either path parameters or query parameters it is possible to specify default parameters. These parameters are used as a fallback when generating a link from an outlet without specifying parameters or the params do not exist in the current route.
+
+```ts
+const config = [
+	{
+		path: 'foo/{foo}',
+		outlet: 'foo',
+		defaultParams: {
+			foo: 'bar'
+		}
+	}
+];
+
+// Using router.link to generate a link for an outlet 'foo', will use the default 'bar' value
+router.link('foo')
+```
+
+A default route can be specified using the optional configuration property `defaultRoute` which will be used if the current route doesn't match an existing. Note there can only be one default route configured otherwise an error will be thrown.
+
+#### Registering Additional Routes
+
+Additional routing configuration can be registered with a router instance, either from the root or by specifying an existing outlet name.
+
+```ts
+const additionalRouteConfig = [
+	{
+		path: 'extra',
+		outlet: 'extra'
+	}
+];
+
+// Will register the extra routes from the route
+router.register(additionalRouteConfig);
+
+// Will register the extra routes from `foo` outlet
+router.register(additionalRouteConfig, 'foo');
+```
+
+If the outlet is not found then an error is thrown.
+
+#### Router Context Injection
+
+To make the `router` instance available to the created outlets and other routing components (such as [`Link`](#link)) Routing leverages a core Dojo 2 widget-core concept [Injecting State](https://github.com/dojo/widget-core/blob/master/README.md#injecting-state). The custom injector `RouterInjector` needs to be defined in a registry available to the components for an known `key`.
```

Comment:
"core Dojo 2 widget-core" -> "Dojo 2 widget-core" or "core `@dojo/widget-core`" (idea being to lighten the redundancy